pytensor/tensor/shape_ops.py

```diff
@@ -0,0 +1,541 @@
+        # example when splitting a packed tensor that had its dims expanded before packing (e.g. when packing shapes
+        # (3, ) and (3, 3) to (3, 4)
+        return type_cast(TensorVariable, x.squeeze(axis=axis))
+
+    [axis] = normalize_axis_tuple(axis, x.ndim)  # type: ignore[misc]
+    shape = as_tensor_variable(shape, dtype="int64")  # type: ignore[arg-type]
+
+    split_op = SplitDims(axis=axis)
+    return type_cast(TensorVariable, split_op(x, shape))
+
+
+def _analyze_axes_list(axes) -> tuple[int, int, int]:
+    """
+    Analyze the provided axes list to determine how many axes are before and after the interval to be raveled, as
+    well as the minimum and maximum number of axes that the inputs can have.
+
+    The rules are:
+    - Axes must be strictly increasing in both the positive and negative parts of the list.
+    - Negative axes must come after positive axes.
+    - There can be at most one "hole" in the axes list, which can be either an implicit hole on an endpoint
+      (e.g. [0, 1]) or an explicit hole in the middle (e.g. [0, 2] or [1, -1]).
+
+    Returns
+    -------
+    n_axes_before: int
+        The number of axes before the interval to be raveled.
+    n_axes_after: int
+        The number of axes after the interval to be raveled.
+    min_axes: int
+        The minimum number of axes that the inputs must have.
+    """
+    if axes is None:
+        return 0, 0, 0
+
+    if isinstance(axes, int):
+        axes = [axes]
+    elif not isinstance(axes, Iterable):
+        raise TypeError("axes must be an int, an iterable of ints, or None")
+
+    axes = list(axes)
+
+    if len(axes) == 0:
+        raise ValueError("axes=[] is ambiguous; use None to ravel all")
+
+    if len(set(axes)) != len(axes):
+        raise ValueError("axes must have no duplicates")
+
+    first_negative_idx = next((i for i, a in enumerate(axes) if a < 0), len(axes))
+    positive_axes = list(axes[:first_negative_idx])
+    negative_axes = list(axes[first_negative_idx:])
+
+    if not all(a < 0 for a in negative_axes):
+        raise ValueError("Negative axes must come after positive")
+
+    def strictly_increasing(s):
+        return all(b > a for a, b in pairwise(s))
+
+    if positive_axes and not strictly_increasing(positive_axes):
+        raise ValueError("Axes must be strictly increasing in the positive part")
+    if negative_axes and not strictly_increasing(negative_axes):
+        raise ValueError("Axes must be strictly increasing in the negative part")
+
+    def find_gaps(s):
+        """Return positions where b - a > 1."""
+        return [i for i, (a, b) in enumerate(pairwise(s)) if b - a > 1]
+
+    pos_gaps = find_gaps(positive_axes)
+    neg_gaps = find_gaps(negative_axes)
+
+    if pos_gaps:
+        raise ValueError("Positive axes must be contiguous")
+    if neg_gaps:
+        raise ValueError("Negative axes must be contiguous")
```

Comment:
You don't  this anywhere? In that case you can change the helper to return a boolean already instead of the "gaps"
```suggestion
    if find_gaps(positive_axes):
        raise ValueError("Positive axes must be contiguous")
    if find_gaps(negative_axes):
        raise ValueError("Negative axes must be contiguous")
```